modules/component-store/src/component-store.ts

```diff
@@ -83,9 +83,8 @@ export class ComponentStore<T extends object> implements OnDestroy {
   private isInitialized = false;
   // Needs to be after destroy$ is declared because it's used in select.
   readonly state$: Observable<T> = this.select((s) => s);
-  // Signal of state$
   readonly state: Signal<T> = toSignal(
-    this.state$.pipe(takeUntil(this.destroy$)),
+    this.stateSubject$.pipe(takeUntil(this.destroy$)),
```

Comment:
The `state$` observable is derived from `stateSubject$` with additional operators such as `shareReplay`, `distinctUntilChanged`, and `map` which is unnecessary in the case of state signal.